### DISCUSSION TITLE
Adding 1st party support for `Product Variant Images` in v2

### DISCUSSION THREAD

**Discussion by bayraak** [medusajs/medusa#9734] [Feature Requests]
Hi,

Currently (including v2.0) there is no support for product variant images (as far as I know). I did dig in the docs and DB schema but definitely it is not supported.

It is a pretty common use-case in e-commerce sites product variants to have images (literally color is a product variant and ofc will look different right? :D)

I think it is a very high priority thing to have it supported 1st party. What are your thoughts?

**Comment by Pirat667** [comment]
Really? That's a basic functionality in most industries like fashion, furniture, etc.

**Reply by rdr6000** [reply]
Yeah. Category, Collection and Variant images are pretty standard in most industries and e-commerce websites

**Comment by joekendal** [comment]
In the end we went with creating a product for each color. And just used variants for sizing.

They can be linked by creating a custom ProductGroup entity.

**Reply by joekendal** [reply]
If you have the images at the variant level, then you'd have duplicates because it creates a variant for each combo of color and size.

So really it is probably best not to have images at the variant level unless they can be linked which would complicate things further

**Comment by joekendal** [comment]
Duplicate #9434
